pmjs

```diff
@@ -0,0 +1,182 @@
+
+Press Ctrl+C to abort current expression, Ctrl+D to exit the REPL`
+};
+
+cmds.exit = python.exit;
+
+/**
+ * Handle a .XXX command. Invokes function cmds[XXX], passing arguments that the user typed
+ * as the function arguments. The function arguments space-delimited arguments; arguments
+ * surrounded by quotes can include spaces, similar to how bash parses arguments. Argument
+ * parsing cribbed from stackoverflow user Tsuneo Yoshioka, question 4031900.
+ *
+ * @param {string} cmdLine     the command the user typed, without the leading .
+ * @returns {string} to display
+ */
+globalThis.replCmd = function replCmd(cmdLine)
+{
+  const cmdName = (cmdLine.match(/^[^ ]+/) || ['help'])[0];
+  const args = cmdLine.slice(cmdName.length).trim();
+  const argv = args.match(/\\\\?.|^$/g).reduce((p, c) => {
+        if (c === '"')
+          p.quote ^= 1;
+        else if (!p.quote && c === ' ')
+          p.a.push('');
+        else
+          p.a[p.a.length-1] += c.replace(/\\\\(.)/,"$1");
+        return  p;
+    }, {a: ['']}).a;
+
+  if (!cmds.hasOwnProperty(cmdName))
+    return `Invalid REPL keyword`;
+  return cmds[cmdName].apply(null, argv);
+}
+
+/** Return String(val) surrounded by appropriate ANSI escape codes to change the console text colour. */
+function colour(colourCode, val)
+{
+  const esc=String.fromCharCode(27);
+  return `${esc}[${colourCode}m${val}${esc}[0m`
+}
+
+/** 
+ * Format result more intelligently than toString. Inspired by Node.js util.inspect, but far less 
+ * capable.
+ */
+globalThis.formatResult = function formatResult(result)
+{
+  switch (typeof result)
+  {
+    case 'undefined':
+      return colour(90, result);
+    case 'function':
+      return colour(36, result);
+    case 'string':
+      return colour(32, `'${result}'`);
+    case 'boolean':
+    case 'number':
+      return colour(33, result);
+    case 'object':
+      if (result instanceof Date)
+        return colour(35, result.toISOString());
+      if (result instanceof Error)
+      {
+        const error = result;
+        const LF = String.fromCharCode(10);
+        const stackEls = error.stack
+                         .split(LF)
+                         .filter(a => a.length > 0)
+                         .map(a => `    ${a}`);
+        return (`${error.name}: ${error.message}` + LF
+          + stackEls[0] + LF
+          + colour(90, stackEls.slice(1).join(LF))
+        );
+      }
+      return JSON.stringify(result);
+    default:
+      return colour(31, `<unknown type ${typeof result}>${result}`);
+  }
+}
+
+/**
+ * Evaluate a complete statement, built by the Python readline loop.
+ */
+globalThis.replEval = function replEval(statement)
+{
+  const indirectEval = eval;
+  try
+  {
+    const result = indirectEval(`${statement}`);
+    return formatResult(result);
+  }
+  catch(error)
+  {
+    return formatResult(error);
+  }
+}
+""");
+
+readline.parse_and_bind('set editing-mode emacs')
+histfile = os. path. expanduser('~/.pmjs_history')
```

Comment:
Is there a typo here? I think it's meant to be `os.path.expanduser('~/.pmjs_history')` right?

Reply:
```suggestion
histfile = os.path.expanduser('~/.pmjs_history')
```